src/psyclone/psyir/nodes/array_of_structures_reference.py

```diff
@@ -96,11 +96,16 @@ def create(symbol, indices, members, parent=None):
                 "The 'symbol' argument to ArrayOfStructuresReference.create() "
                 "should be a DataSymbol but found '{0}'.".format(
                     type(symbol).__name__))
-        if not isinstance(symbol.datatype, symbols.ArrayType):
+        if isinstance(symbol.datatype, symbols.ArrayType):
+            base_type = symbol.datatype.intrinsic
+        elif isinstance(symbol.datatype, (symbols.DeferredType,
```

Comment:
Update the method docstring to say that DeferredType and UnknownType are accepted

Reply:
Done.